sdk/python/feast/feature_server.py

```diff
@@ -250,11 +248,12 @@ async def health():
         )
 
     @app.post("/materialize", dependencies=[Depends(inject_user_details)])
-    def materialize(body=Depends(get_body)):
-        request = MaterializeRequest(**json.loads(body))
-        for feature_view in request.feature_views:
+    def materialize(request: MaterializeRequest) -> None:
+        for feature_view in request.feature_views or []:
+            # TODO: receives a str for resource but isn't in the Union. is str actually allowed?
```

Comment:
it's not allowed, so we should fetch the `feature_view` instance as we do in the `write-to-online-store` endpoint.
let's open a GH issue for that, unless you want to fix it here

Reply:
BTW: thanks for catching it!

Reply:
nah, not here. that should be a sep issue and `fix:` pr

Reply:
https://github.com/feast-dev/feast/issues/4726

Reply:
i'll take care of #4726 once this pr is merged